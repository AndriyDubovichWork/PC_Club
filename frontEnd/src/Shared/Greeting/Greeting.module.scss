.BackGround {
  position: absolute;
  top: 0;
  right: 0;
  z-index: -1;
}

.Greeting {
  width: 80%;
  margin: 0 auto;

  height: 90vh;

  position: relative;

  .PCImage {
    width: 80vh;

    position: absolute;
    right: 0;

    top: 0;
  }
  .Content {
    z-index: 3;

    position: absolute;
    top: 40%;
    transform: translateY(-50%);

    .preTitle {
      font-size: clamp(1.2rem, 1vw, 2rem);
    }
    .title {
      margin: 1vh 0;

      font-size: clamp(1.5rem, 2vw, 3rem);
      font-weight: 700;
      .HighLighted {
        color: #ff4d30;
      }
    }
    .subtitle {
      margin: 1vh 0;
      font-size: clamp(0.75rem, 0.5vw, 1rem);
    }
    .BookButton {
      margin: 0 0.5vw;
    }
    .MoreButton {
      margin: 0 0.5vw;
    }
  }
}
