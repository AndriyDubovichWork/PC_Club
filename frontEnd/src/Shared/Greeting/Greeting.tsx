import React, { useContext } from 'react';
import style from './Greeting.module.scss';
import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
import CheckCircleIcon from '@mui/icons-material/CheckCircle';
import Button from '../../Styled/Button/Button';
import ButtonMUI from '@mui/material/Button';
import { RefContext } from '../../Context/RefContext';
import { useLocation } from 'react-router-dom';
import useIsPCScreen from '../../Hooks/useIsPCScreen';

const Greeting = () => {
  const { BookRef, scrollToElement } = useContext(RefContext);

  const { isPC, textAlign } = useIsPCScreen();
  const location = useLocation();

  const contentWidth = isPC ? '50%' : '100%';

  return (
    <>
      {isPC &&
      (location.pathname === '/PC_Club' ||
        location.pathname === '/PC_Club/') ? (
        <img
          loading='lazy'
          src={require('./assets/backGround.png')}
          className={style.BackGround}
        />
      ) : (
        ''
      )}
      <div className={style.Greeting} style={{ textAlign }}>
        <div className={style.Content} style={{ width: contentWidth }}>
          <h2 className={style.preTitle}>Plan Your day now</h2>
          <div className={style.title}>
            Save <span className={style.HighLighted}>big</span> with our PC
            Clubs
          </div>
          <p className={style.subtitle}>
            Best and cheapest PC Clubs in country
          </p>
          <Button
            onClick={() => {
              scrollToElement(BookRef);
            }}
          >
            Book PC
            <CheckCircleIcon sx={{ marginLeft: '1vw' }} />
          </Button>
          <ButtonMUI
            onClick={() => {
              scrollToElement(BookRef);
            }}
            variant='outlined'
            className={style.MoreButton}
            sx={{
              padding: '2vh 2vw',
              backgroundColor: '#000',
              color: '#fff',
              borderColor: '#000',
              '&:hover': {
                borderColor: '#000',
                backgroundColor: '#fff',
                color: '#000',
              },
            }}
          >
            Learn More
            <ArrowForwardIosIcon />
          </ButtonMUI>
        </div>
        {isPC && (
          <img
            loading='lazy'
            src={require('./assets/pcImg.png')}
            className={style.PCImage}
          />
        )}
      </div>
    </>
  );
};

export default Greeting;
