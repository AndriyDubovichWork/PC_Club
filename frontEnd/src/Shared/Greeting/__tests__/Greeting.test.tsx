import { render, screen, cleanup } from '@testing-library/react';
import Greeting from '../Greeting';
import renderer from 'react-test-renderer';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

test('Greeting match snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter initialEntries={['/PC_Club/']}>
        <Routes>
          <Route path='/PC_Club/' element={<Greeting />} />
        </Routes>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test('Greeting contains text', () => {
  render(
    <MemoryRouter initialEntries={['/PC_Club/']}>
      <Routes>
        <Route path='/PC_Club/' element={<Greeting />} />
      </Routes>
    </MemoryRouter>
  );

  expect(screen.getByText('Plan Your day now')).toBeInTheDocument();
  expect(
    screen.getByText('Best and cheapest PC Clubs in country')
  ).toBeInTheDocument();
  expect(screen.getByText('Book PC')).toBeInTheDocument();
  expect(screen.getByText('Learn More')).toBeInTheDocument();
});
