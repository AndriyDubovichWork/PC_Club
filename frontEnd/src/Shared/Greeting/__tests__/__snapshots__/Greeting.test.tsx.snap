// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting match snapshot 1`] = `
<div
  className="Greeting"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <div
    className="Content"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <h2
      className="preTitle"
    >
      Plan Your day now
    </h2>
    <div
      className="title"
    >
      Save 
      <span
        className="HighLighted"
      >
        big
      </span>
       with our PC Clubs
    </div>
    <p
      className="subtitle"
    >
      Best and cheapest PC Clubs in country
    </p>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-127p27p-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Book PC
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-vhvjub-MuiSvgIcon-root"
        data-testid="CheckCircleIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MoreButton css-1kif34l-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Learn More
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ArrowForwardIosIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
        />
      </svg>
    </button>
  </div>
</div>
`;
